File: DONGUK/프로그래머스_WEEK 1/신규 아이디 추천.py
def solution(new_id):
    answer = ''
    new_id = list(new_id)
    
    # 1단계
    for i in range(len(new_id)):
        alpha = new_id[i]
        if 65 <= ord(alpha) <= 90:
            new_id[i] = chr(ord(alpha) + 32)
    
    # 2단계
    for i in range(len(new_id)):
        alpha = new_id[i]
        if 48 <= ord(alpha) <= 57 or 97 <= ord(alpha) <= 122 or alpha == '-' or alpha == '_' or alpha == '.':
            answer += alpha
    
    # 3단계
    while '..' in answer:
        answer = answer.replace('..', '.')
            
    
    # 4단계
    if answer and answer[0] == '.':
        answer = answer[1:]
    
    if answer and answer[-1] == '.':
        answer = answer[:-1]
    
    # 5단계
    if len(answer) == 0:
        answer += 'a'
        
    # 6단계
    if len(answer) >= 16:
        answer = answer[0:15]
    

    if answer[-1] == '.':
        answer = answer[:-1]

    # 7단계
    while len(answer) <= 2:
        tail = answer[-1]
        answer += tail

    return answer

File: JUNHEE/프로그래머스_옛날문제/신규 아이디 추천.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

string solution(string new_id) {
    string answer = "";
    for(int i = 0; i < new_id.size(); i++){
        if(new_id[i] <= 'Z' && 'A' <= new_id[i])
            new_id[i] += 32;
    }
    cout << "1단계" << new_id << endl;
    for(int i = 0; i < new_id.size(); i++){
        if((new_id[i] <= 'z' && 'a' <= new_id[i]) || 
           ('0' <= new_id[i] && new_id[i] <= '9') || 
           new_id[i] == '-' || 
           new_id[i] == '_' || 
           new_id[i] == '.')
            answer += new_id[i];
    }
    cout << "2단계" << answer << endl;
    for(int i = 1; i < answer.size();){
        if(answer[i-1] == '.' && answer[i] == '.'){
            answer.erase(answer.begin() + i);
        }
        else i++;
    }
    cout << "3단계" << answer << endl;
    if(answer[0] == '.') answer.erase(answer.begin());
    if(answer[answer.size() - 1] == '.') answer.erase(answer.begin() + answer.size() - 1);
    cout << "4단계" << answer << endl;
    if(answer.size() == 0) answer += 'a';
    cout << "5단계" << answer << endl;
    if(answer.size() >= 16) answer = answer.substr(0,15);
    if(answer[answer.size() - 1] == '.') answer.erase(answer.begin() + answer.size() - 1);
    cout << "6단계" << answer << endl;
    if(answer.size() <= 2) {
        char last = answer[answer.size() - 1];
        for(int i = 0; i <= 3 - answer.size(); i++)
            answer += last;
    }
    cout << "7단계" << answer << endl;
    return answer;
}
